wire.py

```diff
@@ -33,26 +33,47 @@ def __init__(self, line, pen, state):
         self.addToGroup(arrow1)
         self.addToGroup(arrow2)
 
-        # line only recieves double click events if it is selectable
-        self.setFlag(QGraphicsItem.ItemIsSelectable, True)
-
-    def mouseDoubleClickEvent(self, event):
-        """register mouse press events"""
-
-        # remove bounding box
-        self.setSelected(False)
+    def contextMenuEvent(self, event):
+        menu = QMenu()
+        delete_action = menu.addAction("delete wire")
+        inspect_action = menu.addAction("inspect wire")
+        selected_action = menu.exec_(QCursor.pos())
+        if selected_action == delete_action:
+            self.deleteWire()
+        elif selected_action == inspect_action:
+            pass
 
+    def deleteWire(self):
+        """delete all backend entries associate with connection and remove
+        from scene"""
         parent = self.state.sym_objects[self.child_key[1]]
         child = self.state.sym_objects[self.parent_key[1]]
 
         # confirm deletion
         dialog = deleteWireDialog("Delete connection between " + parent.name +
                                 "." + self.parent_key[2] + " and " + child.name
                                 + "." + self.parent_key[3] + "?")
-
         # if yes, delete
         if dialog.exec_():
-            self.deleteWire()
+            parent = self.state.sym_objects[self.child_key[1]]
```

Comment:
why don't you use the deleteWire() function here?

Reply:
we're calling deleteWire() from the contextMenu instead since it's not based on the double click event anymore